components/suggestion/search_channel_with_permissions_provider.jsx

```diff
@@ -149,7 +153,7 @@ export default class SearchChannelWithPermissionsProvider extends Provider {
             return;
         }
 
-        const channelsAsync = ChannelActions.autocompleteChannelsForSearch(teamId, channelPrefix)(store.dispatch, store.getState);
+        const channelsAsync = this.autocompleteChannelsForSearch(teamId, channelPrefix);
```

Comment:
Can you check if this is correct?  The previous one is from the mattermost-redux action which has `(teamId, channelPrefix)` params but the new one is from `actions/channel_action` which should have `(term, success, error)`.

Reply:
Hello @saturninoabril , if you look at line `107`, the constructor is passed a `channelSearchFunc`, that function is passed from the component consuming this provider and ultimately, it is `autoCompleteChannelsSearch` from mattermost redux.

Reply:
Oh I see.  The confusing part on me was I assumed that the `channelSearchFunc` passed into `SearchChannelWithPermissionsProvider` and `SearchChannelProvider` are the same (which I only realized now that it's different).